src/libsyntax/parse/token.rs

```diff
@@ -371,6 +371,7 @@ pub enum Nonterminal {
     NtGenerics(ast::Generics),
     NtWhereClause(ast::WhereClause),
     NtArg(ast::Arg),
+    NtVis(ast::Visibility),
```

Comment:
Now the comment `// These are not exposed to macros ....` applies to `NtVis`, it's better moved higher in the list.